scipy/stats/tests/test_distributions.py

```diff
@@ -944,6 +944,20 @@ def test_stats(self):
             assert_array_almost_equal(computed, [mean, var, skew, kurt],
                                       decimal=4)
 
+    @pytest.mark.parametrize('c', [0.1, 0.001])
+    def test_rvs(self, c):
+        # Regression test for gh-11094.
+        x = stats.loggamma.rvs(c, size=100000)
+        # Before gh-11094 was fixed, the case with c=0.001 would
+        # generate many -inf values.
+        assert np.isfinite(x).all()
+        # Crude statistical test.  About half the values should be
```

Comment:
why not use kstest or cramervonmises, maybe just on a subset of the data?

Reply:
Good idea.  I tried, but because the CDF function also has problems when `c` is small, both `kstest` and `cramervonmises` fail for a data set like this.  I'm working on improving the handling of small `c` in `_cdf`, `_sf_`, `_ppf` and `_isf`.  One way that seems to work OK for `_cdf` is to fall back to using `integrate.quad` when `x < -700`, but that is slow (and I don't know how robust it will be in general).